networking_ccloud/tools/netbox_config_gen.py

```diff
@@ -400,12 +407,35 @@ def main():
     parser.add_argument("-r", "--region", required=True)
     parser.add_argument("-v", "--verbose", action="store_true")
     parser.add_argument("-u", "--switch-user", help="Default switch user", required=True)
-    parser.add_argument("-p", "--switch-password", help="Default switch password", required=True)
+    parser.add_argument("-p", "--switch-password", help="Default switch password")
+    parser.add_argument('-V', '--vault-ref',
+                        help="Instead of a password, use this vault references, formatted like <path>:<field>")
+
+    parser.add_argument('-w', '--wrap-in', help='Keys under which the generated config should be nested under. '
+                                                'Format should be: <key1>/<key2>...')
     parser.add_argument("-s", "--shell", action="store_true")
     parser.add_argument("-o", "--output")
 
     args = parser.parse_args()
 
+    if args.switch_password and args.vault_ref:
+        parser.error("You may only use one of '--vault-ref' or '--switch-passsword'")
+    if not (args.switch_password or args.vault_ref):
+        parser.error("Either'--vault-ref' or '--switch-passsword' must be set")
+
+    if args.vault_ref:
+        args.switch_password = VAULT_REF_REPLACEMENT
+        m = re.match(r'^(\S+)\W*:\W*(\S+)', args.vault_ref)
```

Comment:
Could add names to the catchgroups (`?P<path>`, `?P<field>`)

Reply:
but its just to and I did not see a large benefit here, except the regex getting longer.

Reply:
It's documenting the regex. With one view of the regex you know what these () are meant for.